src/util/renderWithUnmount.ts

```diff
@@ -29,7 +30,16 @@ const renderWithUnmount = (
   const unmountObserver = new MutationObserver((ms, observer) => {
     const parentRemoved = ms
       .flatMap((m) => Array.from(m.removedNodes))
-      .some((n) => n === p || n.contains(p));
+      .some((n) => {
+        const htmlEl = n as HTMLElement;
+        const roamBodyRemoved = htmlEl.classList.contains("roam-body-main");
```

Comment:
this is for when renderWithUnmount is used at the page level, eg, `<TldrawCanvas>`

Reply:
I think we could always just say `el.contains(p)`

Reply:
Do you mean `htmlEl.contains(p)`?  This would be the same as it currently is with `n.contains(p)`.
The problem with `n.contains(p)` is that it fires when a preceding sibling is added to an ancestor block, unmounting the component.

![component-container-issue](https://github.com/RoamJS/roamjs-components/assets/3792666/25742a50-2b90-442f-b25e-8ff31aa93bd2)